### DISCUSSION TITLE
JK-PB2A16S20P Support

### DISCUSSION THREAD

**Discussion by silicoon84** [syssi/esphome-jk-bms#447] [General]
I have some of the older JK BMS that i've been using with this project (thanks a million). 
I've recently bought one of the new JK inverter BMS's, which doesn't have the GPS port. It has CAN and RS485 RJ45 ports instead. The RS485 seems like it might not be TTL anymore. [https://www.aliexpress.com/item/1005006020100013.html](url)
I'm still keen to get BMS level detail from it - Any chance you've looked into supporting these boards? 

I've built an ESP32 with BLE and it works, but i don't find its particularly stable. It might be because i had to change the ESP framework type to arduino, as otherwise it wouldnt compile. I read that it an issue with compiling it on an ARM platform. Will try with Windows.
Update: I have tried it on windows and it compiles now. Stability will take a bit to test, but generally not a fan of BLE for things like this, so let me know!

Sorry for the location for the post. Its not exactly an issue given its a new type of BMS, so wasn't sure where to put it!

**Comment by TEETH33** [comment]
HI, I'm interested if you give me some information or a guide on what you've tried on Windows (since it seems to only work with that OS for now) if you can send me the solution step by step with Windows, I have the BMS JK PB2A16S15P and I would like be able to view the data remotely by combining an esphome with a Home assistant

I followed this youtube channel OFF GRID GARAGE
AN Australian who puts these latest bms to the test is called ANDY

**Comment by silicoon84** [comment]
I watch Andy too - great content. 

The reason why it wont build for me, is because i'm running ESPHome/Home Assistant on a raspberry pi and the libraries don't work on that architecture. 

If that's the same problem you're having then basically you just need to install ESPHome on windows `https://esphome.io/guides/installing_esphome.html` and then start the ESPHome dashboard (so you can access it easily)  `https://esphome.io/guides/getting_started_command_line#bonus-esphome-dashboard`

The dashboard should then be accessible from `http://localhost:6052 `

If you're familiar with ESPhome, then just create a configuration as you normally would (follow the instructions in the repository and use bluetooth (ble) example), and flash your device. 

Hope that helps.

**Reply by TEETH33** [reply]
Ok thank you very much and I'll get to work Unfortunately they are at my first experiences ,And besides, I would miss the file to configure,ESP32 A HOMASS

**Reply by silicoon84** [reply]
Have you got home assistant and esphome installed on something? If so, do you have any existing esphome devices?

**Reply by TEETH33** [reply]
notesphome installed That would be fine?
![IMG_20240228_095210](https://github.com/syssi/esphome-jk-bms/assets/160422685/c141539a-f485-4114-8947-1c219db9eac8)
![IMG_20240228_095037](https://github.com/syssi/esphome-jk-bms/assets/160422685/46cff335-c735-47ab-b571-4604eaf3d6d7)

**Comment by txubelaxu** [comment]
I am developing this code:

https://github.com/txubelaxu/esphome-jk-bms/blob/main/components/jk_rs485_bms/README.md
